odoo/addons/base/models/ir_qweb.py

```diff
@@ -2214,8 +2403,51 @@ def _get_asset_nodes(self, bundle, css=True, js=True, debug=False, async_load=Fa
         else:
             return self._generate_asset_nodes_cache(bundle, css, js, debug, async_load, defer_load, lazy_load, media)
 
-    # other methods used for the asset bundles
+    # qweb cache feature
 
+    def _get_cache_key(self, cache_key):
+        """
+            Convert the template cache key item into a hashable key.
+            @params cache_key: tuple
+            @returns: tuple of hashable items
+        """
+        if not isinstance(cache_key, (tuple, list)):
+            cache_key = (cache_key,)
+        keys = []
+        for item in cache_key:
+            try:
+                # use try catch instead of isinstance to detect lazy values
+                keys.append(item._name)
+                keys.append(tuple(item.ids))
+                dates = item.mapped('write_date')
+                if dates:
+                    keys.append(max(dates).timestamp())
+            except AttributeError:
+                keys.append(str(item))
```

Comment:
```suggestion
                keys.append(repr(item))
```

out of caution, because e.g. `str(12) == str("12")`